src/main/webapp/3Dscript.js

```diff
@@ -45,144 +49,285 @@ function initThreeJs() {
 
 
 /**
- * This function fetchs pose data from the RunInfo servlet,
- * it is an asynchronous call requiring addMap() to be 
+ * Fetchs pose data from the RunInfo servlet,
+ * it is an asynchronous call requiring addMap() to be
  * called after the data is fully loaded.
  */
 function fetchData() {
-    const queryString = window.location.search;
-    const urlParams = new URLSearchParams(queryString);
-    const id = urlParams.get('id');
-    const type = urlParams.get('dataType');
-    fetch('/getrun?id=' + id + '&dataType=' + type)
-    .then(response => response.json())
-    .then(data => pose = data)
-    .then(() => addMap());
+  const queryString = window.location.search;
+  const urlParams = new URLSearchParams(queryString);
+  const id = urlParams.get('id');
+  const type = urlParams.get('dataType');
+  fetch('/getrun?id=' + id + '&dataType=' + type)
+      .then((response) => response.json())
+      .then((data) => pose = data)
+      .then(() => addMap());
+}
+
+/**
+ * Finds the median value of the latitude, longitude,
+ * and altitude. The median is used as a reference point
+ * for the East-North-Up (ENU) conversion and as the center
+ * point for the 2D map.
+ * @return {array} An array containing median lat, lng, and alt
+ *   coordinates.
+ */
+function findMedians() {
```

Comment:
Why not just choose an arbitrary point as the ENU origin? It's simpler, and you don't know anything about the shape of the trajectory, so it's hard to say whether the median will always be a better choice.

Reply:
+1
Or you can simply get the four corners of the trajectory instead sort(since it is 2D map, you don't need alt): north east, north west, south east and south west, and then you can compute the center point easily.

Reply:
@jmchristen I used the median mainly because of alignment issues with the 2D map. The Maps Api requires a center point and the ENU conversion requires a reference origin point, the median is the quickest way I could think to align the two. The Maps Static Api also doesn't allow you to specific radius, so while I find a way to generate large 3D terrain, I used median to cover as many points as possible. 
@zjwoody  How would you suggest doing this for a line? I made this as a temporary solution until I could replace the map with terrain, would the center point be a better option?

Reply:
If the radius is small enough to be a problem in practice, then this is okay. (If it's not a problem in practice, I would simplify by just choosing the first point.) My first thought would be to get the center like Jie suggested, but I can't actually think of a strong reason to prefer it over your implementation, other than it's easier to explain.

Reply:
e.g. you have an array of 2D(lat, lng) points, then you can easily find a bounding box of these points, so the box has four corner, it is easy to get the center = ((northest_lat+southest_lat)/2, (eastest_lng+westest_lng)/2). If it is a line(it means two corners are overlapped with the other two), you can still get the correct center.